dali/operators/imgcodec/image_decoder.h

```diff
@@ -546,9 +545,28 @@ class ImageDecoder : public StatelessOperator<Backend> {
           ParseSample(st->parsed_sample,
                       span<const uint8_t>{static_cast<const uint8_t *>(input_sample.raw_data()),
                                           volume(input_sample.shape())});
-          ROI &roi = rois_[i] = GetRoi(spec_, ws, i, st->parsed_sample.dali_img_info.shape);
-          OutputShape(st->out_shape, st->parsed_sample.dali_img_info, format_,
-                      is_planar_layout, use_orientation_, roi);
+          st->out_shape = st->parsed_sample.dali_img_info.shape;
+          st->out_shape[2] = NumberOfChannels(format_, st->out_shape[2]);
+          if (use_orientation_ &&
+              ((st->parsed_sample.nvimgcodec_img_info.orientation.rotated / 90) & 1)) {
```

Comment:
(Nitpick) This `&` is quite non-obvious here. We just want to know if it's rotated by 90 or 270 degrees, why not just
```suggestion
              (st->parsed_sample.nvimgcodec_img_info.orientation.rotated % 180 != 0) {
```